src/device.jl

```diff
@@ -77,30 +77,40 @@ function devices(f::Function, devlist::Union{Integer,AbstractVector})
 end
 
 function filter_free(devlist)
-    usenvml = true
-    if Libdl.find_library(["libnvidia-ml"], []) == ""
-        warning("NVML not found, resorting to nvidia-smi")
+    nvml_libdir = is_windows() ? joinpath(ENV["ProgramFiles"], "NVIDIA Corporation", "NVSMI") : ""
+    if is_windows() && !isdir(nvml_libdir)
+        error("Could not determine Nvidia driver installation location.")
+    end
+    const libnvml = Libdl.find_library(["libnvidia-ml", "nvml"], [nvml_libdir])
```

Comment:
should these be setup before hand or is it fine to just set them up in a function